### PR TITLE
PROD-3120: populate commit and branch for manually triggered github actions

### PR DESCRIPTION
**Pull request by nitishkumar71**
These changes will populate commit and branch information for manually triggered github actions too. Earlier, these informations were only being populated for event triggered actions

### CODE REVIEW

**Review by jossemargt-cto-ai — APPROVED** [review]

File: src/run.js [hunk 1]
```diff
@@ -50,27 +50,39 @@ const getSha = (eventName, payload) => {
   return null
 }
 
+const getBranchFromGithubRef = (ref) => {
+  if (ref.startsWith('refs/')) {
+    const tokens = ref.split('/')
+    return (tokens.length === 3 ? tokens[2] : ref)
+  }
+  return ref
+}
```
Comment (NickAnderegg) [inline src/run.js]: If we're simply returning the original ref if it's not the right format to extract a branch, logging an error message that says explains *why* the original `ref` is being returned will prevent some debugging headaches down the road.

**Review by NickAnderegg — COMMENTED** [review]
LGTM, other than one comment about where we should log an error message to make things easier to debug for users down the road.

File: src/run.js (hunk 1 above)
Comment (nitishkumar71) [reply src/run.js]: Done. I have kept it warn, as `ref` usually would not be in the desired format for `merge` events. So, it's not a error just a warning or info.

File: src/run.js [hunk 2]
```diff
@@ -50,27 +50,40 @@ const getSha = (eventName, payload) => {
   return null
 }
 
+const getBranchFromGithubRef = (ref) => {
+  const tokens = ref.split('/')
+  if (ref.startsWith('refs/') && tokens.length > 2) {
```
Comment (marcoguillenuio) [inline src/run.js:55]: should not be tokens.length >= 2 to allow the case with only 3 tokens?

Comment (nitishkumar71) [reply src/run.js:55]: If you look `ref` field in the context [document](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context). There can be two format i.e. `refs/heads/<branch_name>` and `refs/pull/<pr_number>/merge`. In both case, we want third value in the expression always.

Comment (marcoguillenuio) [reply src/run.js:55]: Got it

**Review by marcoguillenuio — APPROVED** [review]

**Pull request merged**